test/e2e_test.go

```diff
@@ -348,6 +348,31 @@ livez check passed
 			Expect(diff).To(BeEmpty(), "Unexpected metrics")
 		}
 	})
+	It("skip scrape metrics about  nodes with label node-selector filtered in cluster", func() {
```

Comment:
Cleaner test setup would be to add another node to kind cluster that already has the label, however that can be done in followup.